src/crimson/admin/osd_admin.cc

```diff
@@ -157,4 +158,34 @@ class AssertAlwaysHook : public AdminSocketHook {
 };
 template std::unique_ptr<AdminSocketHook> make_asok_hook<AssertAlwaysHook>();
 
+/**
+* A Seastar admin hook: fetching the values of configured metrics
+*/
+class SeastarMetricsHook : public AdminSocketHook {
+public:
+ SeastarMetricsHook()  :
+   AdminSocketHook("perf dump_seastar",
+      "",
+      "dump current configured seastar metrics and their values")
+ {}
+ seastar::future<tell_result_t> call(const cmdmap_t& cmdmap,
+             std::string_view format,
+             ceph::bufferlist&& input) const final
+ {
+   std::unique_ptr<Formatter> f{Formatter::create(format, "json-pretty", "json-pretty")};
+   f->open_object_section("perf_dump_seastar");
+   for (const auto& mf : seastar::metrics::impl::get_value_map()) {
+     for (const auto& m : mf.second) {
+       if (m.second && m.second->is_enabled()) {
+         auto metric_function = m.second->get_function();
```

Comment:
```suggestion
         auto& metric_function = m.second->get_function();
```
no need to have a copy for calling it.